canonical-websites/snapcraft.io
# React test setup with vanilla-framework-react

Fixes #820 

Sets up React and vanilla-framework-react for release page.

Because of issues with running webpack in our current docker image this is still using our current rollup build temporarily.

Rollup itself (and its commonjs plugin) have some issues converting commonjs modules to rollup-friendly modules, so workarounds in config are needed (like explicitly listing named exports needed), so we should move to webpack build as soon as possible.
Also, building JS now takes quite a while...

### QA

- ./run (clean may be needed to update/install all new deps)
- go to (hidden) release page of a snap (http://localhost:8004/account/snaps/lukewh-test/release)
- There should be a 'Hello World' button (rendered by React)
- clicking on it should show an alert

<img width="1028" alt="screen shot 2018-06-27 at 07 06 08" src="https://user-images.githubusercontent.com/83575/41953904-9a7bc89e-79d8-11e8-9165-7f7018ec9753.png">
